component/core/src/main/java/org/exoplatform/social/core/listeners/GroupSynchronizationSocialProfileListener.java

```diff
@@ -102,7 +104,7 @@ private void profileUpdated(ProfileLifeCycleEvent profileLifeCycleEvent) {
     }
   }
 
-  private void synchronizePropertyGroups(Profile profile, User user) {
+  private static synchronized void synchronizePropertyGroups(Profile profile, User user) {
```

Comment:
Why put all in static ? 
Im not sure it is the correct way to do

Reply:
if we put just synchronized, it will be  applied only in the thread instance, for multi threading we need to make it static so we synchronize the execution of the calss instance.

Reply:
We use synchronized nowhere in a listener, nor static in fact. 

I think we should manage the synchronized in the service level, but not in the listener. 
As we have only one instance of the service, using synchronized should be ok.

Reply:
As discussed, putting the updateProfile out of the loop avoid having the problem of concurrent changes on organization service, so no ned to synchronize the groups operation